README.md

```diff
@@ -1,70 +1,242 @@
+    -   VPC; Select VPC to Build in
+    -   SSHSecurityGroup; Select group for SSH access
+
+#### Options for Larger Setups
+These are the items to best look when round 5k Nodes checking in
+
+-   Params to Change only if 4-5k+ Nodes (Estimated; still need to do actual load tests)
+    -   DBInstanceType; Default (t2.large) is the minimum required for Chef DB Connections
+    -   ESProxyInstanceType; Default is good for small-medium Chef Build Outs. Change as needed
+    -   ElasticInstanceType; Default is probably not going to cut it for tons of requests
+    -   DBMultiAZ; Would recommend setting this to true
+
+#### Optional Setup
+
+-   Restoring Setups; See the [Restore/Backup](#restorebackup-options) Section
+-   NewRelic Setup; See the [NewRelic](#new-relic) Section
+-   Sumologic Setup; See the [Sumologic](#sumologic) Section
+
+##### Mail Server Setup
+If you fill out the following Params the script will setup a postfix relay to allow the Chef Server to send email.
+
+-   MailCreds; if the format `$username:$password`
+-   MailHost; Such as mailgun.org
+-   MailPort; Mail server port
+
+##### Development Selections
+
+Options that don't need to be bothered with except in development situations.
+
+See [Contributing](#contributing) for how to contribute to this project
+
+-   GitBranch; sets branch to pull scripts from
+    -   Currently only affects NewRelic and Sumologic
+
+### SSL Setup
+
+To simplify the setup only the ELB (Public Side) is setup with SSL.
+
+Learn how to upload certificates to be used with this [Guide](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_server-certs.html)
+
+To Get an existing SSL Cert in AWS Follow this [Guide](http://docs.aws.amazon.com/cli/latest/reference/iam/get-server-certificate.html)
+
+or run this command `aws iam get-server-certificate --server-certificate-name`
+
+You can also use the new [AWS Certificate Manager Service](http://docs.aws.amazon.com/acm/latest/userguide/gs-acm.html) to create and manage certificates
+
+### Blue/Green Deployment
+
+This setup supports doing staged deploys and/or blue/green deployments
+
+Below describes how this is accomplished, but there is a manual step require to complete the deploy.
+
+-   Manual step
+    -   Change `chef.$HostedZone` in Route53 to/from `chef-a.$HostedZone` or `chef-b.$HostedZone`
+
+
+-   Blue/Green Options:
+    -   ChefSubdomain Param; Enabled having a chef-a, chef-b, and even a chef-test subdomain created
+        -   Subnets are changed based on these settings
+    -   Server names are created based on the Subdomain and a prime chef.domain.com to support the multi-dns setup; [Code](aws_backendless_chef_ha.yml#L1149)
+    -   There is a workaround for the odd cert creation caused by the multiple server name setup;
+        -   Specify Cert Location: [Code 01](aws_backendless_chef_ha.yml#L1147-L1148)
+        -   Copy badly name cert to expected one: [Code 02](aws_backendless_chef_ha.yml#L1194-L1200)
```

Comment:
Links to code lines like this are likely to float.  Any ideas on how we could anchor this better or provide the code inline here?